### PR TITLE
Add note about frequent RTT estimates

### LINKED ISSUE #2067: Samples for RTT estimation
Based on the recovery draft, a largest newly acked packet should be considered as sample for the RTT estimation.

This means that if the receiver sends 1 ACK for every packet, then every single ACK will be taken as sample for RTT estimation. But RFC6298 suggests to take at least one RTT measurement per RTT and also states, "Additionally, when multiple samples are taken per RTT, the alpha and beta defined in Section 2 may keep an inadequate RTT history."

I think taking every single largest ACK into consideration doesn't comply with RFC6298.

### PR DESCRIPTION
**Pull request by janaiyengar**
Closes #2067.

### CODE REVIEW

**Comment by erickinnear** [comment]
Looks good to me, always nice to pull that note out of the reference and make sure people see it.

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -292,6 +292,10 @@ ack delay field should be ignored.
 Like TCP, QUIC calculates both smoothed RTT and RTT variance similar to those
 specified in {{?RFC6298}}.
 
+A sender might take multiple samples per RTT, and as suggested in {{?RFC6298}},
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]:
```suggestion
Taking a sample from every ACK with a larger largest acked is likely to cause a sender to take multiple samples per RTT, and as suggested in {{?RFC6298}},
```

Comment (janaiyengar) [reply draft-ietf-quic-recovery.md]: I've rephrased, take a look

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -289,8 +289,14 @@ be subtracted from the RTT as long as the result is larger than the min_rtt.
 If the result is smaller than the min_rtt, the RTT should be used, but the
 ack delay field should be ignored.
 
-Like TCP, QUIC calculates both smoothed RTT and RTT variance similar to those
-specified in {{?RFC6298}}.
+A sender calculates both smoothed RTT and RTT variance similar to those
+specified in {{?RFC6298}}, see {{on-ack-received}}.
+
+A sender will commonly receive ACK frames that acknowledge larger packet numbers
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: Please add text on why this is relevant.  Suggestion "that increase largest acked and therefore generate an RTT sample."

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -289,8 +289,14 @@ be subtracted from the RTT as long as the result is larger than the min_rtt.
...
+than before.  A sender might therefore take multiple samples per RTT (see
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: Suggestion: replace "A sender..." on with: "When multiple samples are generated within an RTT, the smoothed RTT may retain inadequate RTT history, as suggested in {{?RFC6298}}.  Utilizing multiple RTT samples per RTT in smoothed RTT and RTT variance in a different way is currently an open research question."

File: draft-ietf-quic-recovery.md (hunk 2 above)
Comment (janaiyengar) [reply draft-ietf-quic-recovery.md]: I've taken your suggestion but rephrased to avoid a run-on sentence.

File: draft-ietf-quic-recovery.md (hunk 1 above)
Comment (ianswett) [reply draft-ietf-quic-recovery.md]: I really am not in love with the "A sender might" start.

File: draft-ietf-quic-recovery.md [hunk 4]
```diff
@@ -289,8 +289,15 @@ be subtracted from the RTT as long as the result is larger than the min_rtt.
 If the result is smaller than the min_rtt, the RTT should be used, but the
 ack delay field should be ignored.
 
-Like TCP, QUIC calculates both smoothed RTT and RTT variance similar to those
-specified in {{?RFC6298}}.
+A sender calculates both smoothed RTT and RTT variance similar to those
+specified in {{?RFC6298}}, see {{on-ack-received}}.
+
+A sender might take multiple samples per RTT because ACK frames that are
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: Suggestion: "A sender will take multiple samples per RTT when ACK frames are received within an RTT that acknowledge larger packet numbers than before("

Comment (martinthomson) [reply draft-ietf-quic-recovery.md]: A sender will take multiple samples per RTT.  RTT samples are taken when the largest acknowledged packet number is increased by an acknowledgment (see).

**Comment by janaiyengar** [comment]
merging, so we can get this into -18

**Pull request merged**